### DISCUSSION TITLE
Bring back the RDFGraphs (or a similar) feature

### DISCUSSION THREAD

**Discussion by semihsalihoglu-uw** [kuzudb/kuzu#4308] [General]
We have decided that starting from 0.7.0, we will remove our RDFGraphs feature (at least temporarily). The original feature was outlined in this [issue](https://github.com/kuzudb/kuzu/issues/1570). This was a feature that allowed users to natively import turtle or n3 files into Kuzu and get basic querying over them in Cypher. 

There were a few justifications behind the hard decision of removing the feature. First, we saw little usage for this native support for RDF. We also saw a few people who were importing RDF data to Kuzu but not through RDFGraphs, but instead they were using their own mappings from RDF To property graph. Second, we realized we were very aggressive in making this a core-feature of Kuzu, instead of an [extension](https://docs.kuzudb.com/extensions/). Implementing this at the core means we have to support several RDF-specific features internally, e.g., the VARIANT data type, or virtual properties (the iri property on the predicates are actually virtual and internally we store integers) and maintaining all these features at the core has been slowing our development.

We can bring this feature back but we should do it once we accumulate more evidence that there is significant demand for a feature like this. And when we do, we should do the hard work of making it an extension, so we can decouple this feature more from the core.

You can weigh in here about your use cases if this is important for you.

**Comment by emabiz** [comment]
In my opinion the most interesting part of the RDF is the inference rule system, that will make the queries very powerful, like transitivity or sameAs relationship.
If there is the possibility to describe this kind of relationships in kuzu, keeping its internal structure, maybe the implementation of a complete RDF db is not needed.
But I am not sure that this is possible without a massive work

**Reply by semihsalihoglu-uw** [reply]
I agree that the inferencing ability of RDF is the most interesting part. It also seems like inferencing is the RDF feature that is hardest to horn into a property graph system. Our goal with RDFGraphs was just the opposite: give general querying of RDF triples except inferencing. 

I don't yet have great ideas about something very practical. Happy to hear about people's thoughts.
